react/src/base/inputs/SprkSelect/SprkSelect.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import uniqueId from 'lodash/uniqueId';
+
+class SprkSelect extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      choiceItems: props.choices.map((item) => ({ id: uniqueId(), ...item })),
```

Comment:
Do we want them to be able to set their own `id` for each choice?

Reply:
@guern1kn , this id is used for the react key="" on each option. I did check MDN and there is no id attr for the option element. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option